src/Razor/test/Microsoft.VisualStudio.LanguageServices.Razor.Test/DocumentGenerator/BackgroundDocumentGeneratorTest.cs

```diff
@@ -382,6 +382,10 @@ public TestDynamicFileInfoProvider()
 
             public IReadOnlyDictionary<string, DynamicDocumentContainer> DynamicDocuments => _dynamicDocuments;
 
+            public override void Initialize(ProjectSnapshotManagerBase projectManager)
```

Comment:
This is a pretty good argument for `ProjectSnapshotChangeTrigger` to be converted to an interface IMO

Reply:
Totally agree! It definitely embodies the "I Can do" relationship. Filed: https://github.com/dotnet/aspnetcore/issues/35930